packages/flutter/lib/src/http/response.dart

```diff
@@ -27,4 +24,67 @@ class Response {
   ///
   /// The code 500 is used when no status code could be obtained from the host.
   final int statusCode;
+
+  /// The headers for this response.
+  final Map<String, String> headers;
+}
+
+bool _isSpace(String c) {
+  return c == ' ' || c == '\t' || c == '\n' || c == '\r' || c == '\f';
+}
+
+int _skipSpaces(String string, int index) {
+  while (index < string.length && _isSpace(string[index]))
+    index += 1;
+  return index;
+}
+
+// https://html.spec.whatwg.org/#algorithm-for-extracting-a-character-encoding-from-a-meta-element
```

Comment:
fwiw I'm not convinced that this is the right algorithm, but from my phone I am having no success in finding the right one, so...